### PR TITLE
[ATS/en] Add documentation for ATS

### PR DESCRIPTION
**Pull request by mb64**
- [X] I solemnly swear that this is all original content of which I am the original author
- [X] Pull request title is prepended with `[language/lang-code]`
- [X] Pull request touches only one file (or a set of logically related files with similar changes made)
- [X] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- (N/A) If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - (N/A) Yes, I have double-checked quotes and field names!

I doubt that the syntax highlighter will support ATS, so this will at the very least require some edits to figure out the best syntax highlighting.  Other than that, if there's anything else I can do to improve it, I will be happy to edit it.

### CODE REVIEW

File: ATS.html.markdown [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+
+// Proof-level lists of ints, of type 'sort'
+datasort IntList = ILNil of ()
+                 | ILCons of (int, IntList)
+
+// ILAppend(x,y,z) iff x ++ y = z
+dataprop ILAppend(IntList, IntList, IntList) =
+  | {x:IntList} AppendNil(x, ILNil, x)
+  | {a:int} {x,y,z: IntList}
+    AppendCons(x, ILCons(a,y), ILCons(a,z)) of ILAppend(x,y,z)
```
Comment (thalesmg) [inline ATS.html.markdown:210]: Shouldn't this be `AppendCons(ILCons(a, x), y, ILCons(a,z)) of ILAppend(x,y,z)` for `x ++ y = z` to hold?
If so, the base clause seems to be flipped as well.

Comment (mb64) [reply ATS.html.markdown:210]: You're absolutely right, I switched the arguments -- thanks for catching it.  I will fix it and rebase the PR.

**Comment by mb64** [comment]
There was some discussion of this PR on the ATS mailing list; [here is a link](https://groups.google.com/g/ats-lang-users/c/nG8f1CdwymI).

File: ATS.html.markdown [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+prfun append_total {x,y: IntList} .<x>. (): [z:IntList] ILAppend(x,y,z)
+  = scase x of // scase lets you inspect static arguments (only in prfuns)
+  | ILNil() => AppendNil
+  | ILCons(a,rest) => AppendCons(append_total())
+
+
+/*************** Part 4: views ****************/
+
+// views are like props, but linear; ie they must be consumed exactly once
+// prop is a subtype of view
```
Comment (d4v3y5c0n3s) [inline ATS.html.markdown:224]: Isn't "view" a subtype of "prop", not the other way around?  Maybe I'm mistaken, but the Intro to ATS book states that view is a linear version of prop.

Comment (d4v3y5c0n3s) [reply ATS.html.markdown:224]: Nevermind, I misunderstood. I'll delete my comments shortly.

**Review by d4v3y5c0n3s — APPROVED** [review]
Having read this pull request, I can say that this is a great reference for the ATS programming language.  I think this would be very useful to anyone who uses or plans to use ATS.  Good luck, and I hope this gets merged. 😎

**Comment by timmyjose** [comment]
Any updates on this? If there anything stopping this PR from getting merged? I'm a beginner in ATS myself, and I find this resource quite useful.

**Comment by vendethiel** [comment]
Thank you!

**Pull request merged**